### PR TITLE
Add JWT authentication example

### PR DESCRIPTION
**Pull request by dimensi0n**
Related to #31

### CODE REVIEW

File: expert/jwt_authentication.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
```
Comment (SimonWaldherr) [inline expert/jwt_authentication.go]: imported and not used: "os"

File: expert/jwt_authentication.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+// We add jwt.StandardClaims as an embedded type, to provide fields like expiry time
+type Claims struct {
+	Username string `json:"username"`
+	jwt.StandardClaims
+}
+
+// Signin handler to handle signin request
+func Signin(w http.ResponseWriter, r *http.Request) {
+
+	jwtKey = []byte("this_is_a_secret_token")
```
Comment (SimonWaldherr) [inline expert/jwt_authentication.go]: undefined: jwtKey

maybe you mean ```jwtKey := []byte("this_is_a_secret_token")```

File: expert/jwt_authentication.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
```
Comment (SimonWaldherr) [inline expert/jwt_authentication.go]: function main is undeclared in the main package

**Comment by dimensi0n** [comment]
Edited

**Comment by SimonWaldherr** [comment]
sorry for the delay.
please take a look at line 69:
```
go run expert/jwt_authentication.go 
# command-line-arguments
expert/jwt_authentication.go:69:29: undefined: controllers
```

**Comment by dimensi0n** [comment]
Fixed

**Comment by SimonWaldherr** [comment]
thanks, now it runs.
Can you please provide a little information how to use it in the readme file?

e.g.:
```
go run expert/jwt_authentication.go
```
then open <http://localhost:8080/signin> etc. ...

**Pull request closed without merging**